virt-v2v/cold/entrypoint.go

```diff
@@ -0,0 +1,260 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+)
+
+const (
+	OVA     = "ova"
+	vSphere = "vSphere"
+	DIR     = "/var/tmp/v2v"
+	FS      = "/mnt/disks/disk[0-9]*"
+	Block   = "/dev/block[0-9]*"
+	VDDK    = "/opt/vmware-vix-disklib-distrib"
+)
+
+var (
+	xmlFilePath string
+	server      *http.Server
+)
+
+func main() {
+	virtV2vArgs := []string{"virt-v2v", "-v", "-x"}
+	source := os.Getenv("V2V_source")
+
+	if isValidSource(source) {
+		fmt.Printf("virt-v2v supports the following providers: {OVA, vSphere}. Provided: %s\n", source)
+		os.Exit(1)
+	}
+
+	requiredEnvVars := map[string][]string{
+		vSphere: {"V2V_libvirtURL", "V2V_secretKey", "V2V_vmName"},
+		OVA:     {"V2V_diskPath", "V2V_vmName"},
+	}
+
+	if envVars, ok := requiredEnvVars[source]; ok {
+		if !checkEnvVariablesSet(envVars...) {
+			fmt.Printf("Following environment variables need to be defined: %v\n", envVars)
+			os.Exit(1)
+		}
+	}
+
+	fmt.Println("Preparing virt-v2v")
+	switch source {
+	case vSphere:
+		virtV2vArgs = append(virtV2vArgs, "-i", "libvirt", "-ic", os.Getenv("V2V_libvirtURL"))
+	case OVA:
+		virtV2vArgs = append(virtV2vArgs, "-i", "ova", os.Getenv("V2V_diskPath"))
+	}
+
+	if err := os.MkdirAll(DIR, os.ModePerm); err != nil {
+		fmt.Println("Error creating directory  ", err)
+		os.Exit(1)
+	}
+	virtV2vArgs = append(virtV2vArgs, "-o", "local", "-os", DIR)
+
+	//Disks on filesystem storage.
+	if err := LinkDisks(FS, 15); err != nil {
+		os.Exit(1)
+	}
+	//Disks on block storage.
+	if err := LinkDisks(Block, 10); err != nil {
+		os.Exit(1)
+	}
+
+	if source == vSphere {
+		if _, err := os.Stat("/etc/secret/cacert"); err == nil {
+			// use the specified certificate
+			err = os.Symlink("/etc/secret/cacert", "/opt/ca-bundle.crt")
+			if err != nil {
+				fmt.Println("Error creating ca cert link ", err)
+				os.Exit(1)
+			}
+		} else {
+			// otherwise, keep system pool certificates
+			err := os.Symlink("/etc/pki/tls/certs/ca-bundle.crt.bak", "/opt/ca-bundle.crt")
+			if err != nil {
+				fmt.Println("Error creating ca cert link ", err)
+				os.Exit(1)
+			}
+		}
+		virtV2vArgs = append(virtV2vArgs, "-ip", "/etc/secret/secretKey")
+
+		if info, err := os.Stat(VDDK); err == nil && info.IsDir() {
+			virtV2vArgs = append(virtV2vArgs,
+				"-it", "vddk",
+				"-io", fmt.Sprintf("vddk-libdir=%s", VDDK),
+				"-io", fmt.Sprintf("vddk-thumbprint=%s", os.Getenv("V2V_fingerprint")),
+			)
+		}
+		virtV2vArgs = append(virtV2vArgs, "--", os.Getenv("V2V_vmName"))
+	}
+
+	if err := executeVirtV2v(virtV2vArgs); err != nil {
+		fmt.Println("Error executing virt-v2v command ", err)
+		os.Exit(1)
+	}
+
+	if source == OVA {
```

Comment:
without the controller part, the conversion pod would remain running, no? in that case, let's drop this part and add it along with the changes to forklift-controller

Reply:
yeah, it wont complete the pod - removing.